kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/TestCaseExecutor.kt

```diff
@@ -78,7 +80,12 @@ class TestCaseExecutor(
    ): TestResult {
       return when {
          extensions.isEmpty() -> executeIfActive(testCase) { executeActiveTest(testCase, context, start) }
-         else -> extensions.first().intercept(testCase) { intercept(it, context, start, extensions.drop(1)) }
+         else -> extensions.first().intercept(testCase) {
+            // the user's intercept method is free to change the context of the coroutine
+            // to support this, we should switch the context used by the test case context
+            val newContext = context.withCoroutineContext(coroutineContext)
```

Comment:
Maybe I'm missing something, but I don't really see why handing down the augmented `coroutineContext` is necessary... from my understanding the fact that `TestContext` extends `CoroutineScope` is just for convenience, so you can `launch` etc. directly in the test without creating a new scope.

The extensions don't receive a `TestContext` parameter, so they can't access or modify the `TestContext.coroutineContext`. And `TestContext.coroutineContext` gets replaced anyway by the current coroutine scope's context just before the actual test is called (which seems to happen in `TestCaseExecutor.executeAndWait`).

Reply:
I guess it was because I wanted the TestContext to be "correct", in that it represented the currently used coroutine context, rather than the one it was started with, which has perhaps been overwritten by the user (using withContext). 

If TestContext isn't exposed to the user until executeAndWait then I guess it doesn't matter, and it can rely on the final switch just before the test is executed. 

But if we add some kind of further extensions later that expose TestContext, them it would have an outdated coroutine context reference.

I think I'm right on the above anyway.